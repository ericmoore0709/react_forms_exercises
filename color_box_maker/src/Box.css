.Box {
    border: 1px solid black;
    margin: auto;
}

.Box>button {
    background-color: red;
    border: none;
    color: white;
}