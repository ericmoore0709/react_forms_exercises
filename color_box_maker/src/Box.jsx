import './Box.css';
import PropTypes from 'prop-types';

const Box = ({ id, color = 'white', width = 50, height = 50, removeBox }) => {

    const handleClick = (e) => {
        const id = e.target.parentNode.id;
        removeBox(id);
    }

    return (
        <div
            id={id}
            className="Box"
            style={{
                backgroundColor: color,
                height: `${height}px`,
                width: `${width}px`
            }}>
            <button onClick={handleClick}>X</button>
        </div>
    )
}

Box.propTypes = {
    id: PropTypes.string.isRequired,
    color: PropTypes.string,
    width: PropTypes.number,
    height: PropTypes.number,
    removeBox: PropTypes.func.isRequired
}

export default Box;