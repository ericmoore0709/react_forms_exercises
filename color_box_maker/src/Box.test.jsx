import { expect, it, vi } from 'vitest'
import { render, fireEvent } from '@testing-library/react';
import Box from './Box';

it('should not smoke', () => {
    render(<Box />);
});

it('should match snapshot', () => {
    const { asFragment } = render(<Box />);
    expect(asFragment()).toMatchSnapshot();
});

it('should fire removeBox when X is clicked', () => {
    const removeMock = vi.fn();
    const { container } = render(<Box id='box-id' removeBox={removeMock} />);
    const button = container.querySelector('button');
    expect(button).toBeInTheDocument();
    fireEvent.click(button);
    expect(removeMock).toBeCalledWith('box-id');
})

it('should render with default props', () => {
    const { container } = render(<Box />);
    const box = container.querySelector('.Box');
    expect(box).toHaveStyle('background-color: #ffffff');
    expect(box).toHaveStyle('width: 50px');
    expect(box).toHaveStyle('height: 50px');
});

it('should render with custom props', () => {
    const { container } = render(<Box id="box-id" color="#ff0000" width={100} height={150} />);
    const box = container.querySelector('.Box');
    expect(box).toHaveStyle('background-color: #ff0000');
    expect(box).toHaveStyle('width: 100px');
    expect(box).toHaveStyle('height: 150px');
});