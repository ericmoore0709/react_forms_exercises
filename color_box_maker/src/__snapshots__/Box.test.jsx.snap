// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="Box"
    style="background-color: white; height: 50px; width: 50px;"
  >
    <button>
      X
    </button>
  </div>
</DocumentFragment>
`;
